lib/miq_expression/remove_equality_visitor.rb

```diff
@@ -0,0 +1,71 @@
+class MiqExpression
+  # rubocop:disable Naming/MethodName
+  class RemoveEqualityVisitor < Arel::Visitors::Reduce
+    def visit_NilClass(object, _collector)
+      object
+    end
+
+    def visit_Arel_Nodes_Equality(object, collector)
+      field, binds = collector
+      if (object.left == field.left && object.right == field.right) ||
+         (object.right == field.left && object.left == field.right)
+        # remove this node from the query
+        nil
+      elsif object.right.kind_of?(Arel::Nodes::BindParam)
```

Comment:
Should this be done in a separate `if` prior to the `if (object.left == ...` stuff above?  I am curious if there would ever be a case where the `Arel::Nodes::Equality` we are looking for would be behind a bind variable?  Is that possible or am I just being stupid?

(you are allowed to answer honestly to that last one)

Reply:
For our use case, we are looking for a join between two tables. these should be 2 id columns, and I can't think of a reason why it would be a bind variable.

The extra scoping around this join is what we are keeping, which often does have a bind variable.

Removing an expression with a bind variable would cause issues. a bind should be popped in this case. I could add this case but it seems kinda out there.